sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -400,33 +436,147 @@ private PAssert() {}
 
   ////////////////////////////////////////////////////////////
 
-  private static class PAssertionSite implements Serializable {
+  /**
+   * Output of {@link PAssert}. Passed to a conclude function to act upon.
+   */
+  @DefaultCoder(AvroCoder.class)
+  public static final class SuccessOrFailure {
+
+    private final boolean isSuccess;
+    @org.apache.avro.reflect.Nullable
+    private final PAssertionSite site;
```

Comment:
If the throwable was serializable with Java serialization, it would be better just to use SerializableCoder. If that isn't the case, it may make sense to create an "ExceptionCoder" that could be used for this case as well as others. Maybe worth creating a JIRA to track that, and marking this TODO(that)?